app/code/Magento/OfflineShipping/Model/Quote/Address/FreeShipping.php

```diff
@@ -65,9 +65,7 @@ public function isFreeShipping(\Magento\Quote\Model\Quote $quote, $items)
             $itemFreeShipping = (bool)$item->getFreeShipping();
             $addressFreeShipping = $addressFreeShipping && $itemFreeShipping;
 
-            if ($addressFreeShipping && !$item->getAddress()->getFreeShipping()) {
-                $item->getAddress()->setFreeShipping(true);
-            }
+            $shippingAddress->setFreeShipping($addressFreeShipping && $itemFreeShipping);
```

Comment:
After the change there are two exact same rows:

```
$addressFreeShipping = $addressFreeShipping && $itemFreeShipping;
 	 
+            $shippingAddress->setFreeShipping($addressFreeShipping && $itemFreeShipping);
```
Why `$item->getAddress()->setFreeShipping(true)` is ignored now?

Please describe steps in more details. As I understand the old logic, it's enough to have one item offering free shipping in order to get free for the address and it looks perfectly valid to me. In new logic it's seems like you try to check ALL items, isn't it?